src/homework/api/application.py

```diff
@@ -1,3 +1,100 @@
-from fastapi import APIRouter
+from uuid import uuid4
+from secrets import token_hex
+import fastapi
+from src.homework.api.contracts import (
+    CreateApplicationRequest,
+    UpdateApplicationRequest,
+    DeleteApplicationRequest,
+    ResponseStatus,
+    ApplicationDataResponse,
+)
+from src.homework.api.security import (
+    ValidateHeader,
+    MasterAccessLevel,
+    issue_token,
+)
+import src.homework.db.crud as crud
 
-router = APIRouter()
+router = fastapi.APIRouter(
+    dependencies=[
+        fastapi.Depends(ValidateHeader(MasterAccessLevel("MasterApp")))
+    ]

```

Comment:
```suggestion
router = fastapi.APIRouter(
    dependencies=[
        fastapi.Depends(ValidateHeader(MasterAccessLevel("MasterApp"))),
        prefix="/application"
    ]

```
Consider adding common prefix for all endpoints to avoid code duplication

Reply:
This is a good point, however, the prefix introduction will create redundancy in the route making it "/application/application". If we change the route inside the router to be "/", requests will be expected at "/application/", which is less intuitive than the current path. Anyway, thanks for paying attention to this aspect!

Reply:
Thank you for your considerateness! I completely agree with you.